Source/ACE.Server/Entity/Actions/DelayManager.cs

```diff
@@ -3,6 +3,8 @@
 
 using log4net;
 
+using ACE.Common;
+
 namespace ACE.Server.Entity.Actions
 {
     public class DelayManager : IActor
```

Comment:
Having a global 'DelayManager' is also an entirely unneeded construct

There is also a core bug here that has not been addressed, where DelayManager is always 1 tick behind. This causes subtle timing bugs all throughout the code.

Reply:
Global DelayManager is a requirement for thread safety.

Even with the thread safety requirement, it's still a convenient and easy thing to have. The global delay manager isn't doing any heavy lifting. It simply manages delayed actions. Once the delay is expired, the NextAct action is enqueued onto the IActor, and the action is run on the IActors thread. Currently Landblocks are single threaded, but, in the future when Landblocks are split up into groups, the actual action will be run on the appropriate landblock group thread.

I'll investigate the tick behind.